### DISCUSSION TITLE
how the overlap sizes combined with the dmlibmesh

### DISCUSSION THREAD

**Discussion by Ricardotu** [libMesh/libmesh#3902] [Q&A]
I'm testing the impact of overlap size on my ASM preconditioner. I noticed in the libMesh source code that there's a `dmlibmesh`, which seems to be the default DM in libMesh. I would like to understand how mesh-based overlap is implemented in libMesh. Also, which part of the code combines the overlap with `dmlibmesh`? I mean if there is mesh-based overlaps

**Comment by roystgnr** [comment]
If you're testing `-pc_asm_overlap`, that's done at the PETSc level independently of libMesh ghosting levels and DM interfaces.

**Reply by Ricardotu** [reply]
Is this on the level of algebra? There is a code comment`/* FIX: allow mesh-based overlap. */`  in `petscdmlibmeshimpl.C` . It doesn't seem to be mesh-based.
